src/Agent.Worker/TestResults/TrxResultReader.cs

```diff
@@ -407,14 +438,112 @@ private List<TestCaseResultData> ReadActualResults(XmlNodeList resultsNodes, str
                     if ((consoleLog = resultNode.SelectSingleNode("./Output/StdOut")) != null && !string.IsNullOrWhiteSpace(consoleLog.InnerText))
                     {
-                        resultCreateModel.ConsoleLog = consoleLog.InnerText;
+                        resultCreateModel.AttachmentData.ConsoleLog = consoleLog.InnerText;
                     }
 
                     // standard error
                     if ((standardError = resultNode.SelectSingleNode("./Output/StdErr")) != null && !string.IsNullOrWhiteSpace(standardError.InnerText))
                     {
-                        resultCreateModel.StandardError = standardError.InnerText;
+                        resultCreateModel.AttachmentData.StandardError = standardError.InnerText;
+                    }
+                }
+
+                XmlNode innerResults = resultNode.SelectSingleNode("InnerResults");
+                if (innerResults != null)
+                {
+                    resultCreateModel.ResultGroupType = GetResultGroupType(resultNode, testType);
+
+                    XmlNodeList resNodes = innerResults.SelectNodes("UnitTestResult");
+                    XmlNodeList webTestResultNodes = innerResults.SelectNodes("WebTestResult");
+                    XmlNodeList orderedTestResultNodes = innerResults.SelectNodes("TestResultAggregation");
+
+                    resultCreateModel.TestCaseSubResultData = new List<TestCaseSubResultData>();
+
+                    resultCreateModel.TestCaseSubResultData.AddRange(ReadActualSubResults(resNodes, TestType.UnitTest, 1));
+                    resultCreateModel.TestCaseSubResultData.AddRange(ReadActualSubResults(webTestResultNodes, TestType.WebTest, 1));
+                    resultCreateModel.TestCaseSubResultData.AddRange(ReadActualSubResults(orderedTestResultNodes, TestType.OrderedTest, 1));
+                }
+
+                lock (sync)
+                {
+                    //Mandatory fields. Skip if they are not available.
+                    if (!string.IsNullOrEmpty(resultCreateModel.AutomatedTestName) && !string.IsNullOrEmpty(resultCreateModel.TestCaseTitle))
+                    {
+                        results.Add(resultCreateModel);
                     }
+
+                }
+            });
+
+            return results;
+        }
+
+        private List<TestCaseSubResultData> ReadActualSubResults(XmlNodeList resultsNodes, TestType testType, int level)
+        {
+            List<TestCaseSubResultData> results = new List<TestCaseSubResultData>();
+
+            object sync = new object();
+            Parallel.ForEach<XmlNode>(resultsNodes.Cast<XmlNode>(), resultNode =>
+            {
+                TestCaseSubResultData resultCreateModel = new TestCaseSubResultData();
+
+                //Find and format dates as per TCM requirement.
+                TimeSpan duration;
+                if (resultNode.Attributes["duration"] != null && resultNode.Attributes["duration"].Value != null)
```

Comment:
Are these case sensitive?

Reply:
xml has a fixed format and its "duration" (case sensitive)